MediaAdmin/FileUtils/Image/ImagickImageManager.php

```diff
@@ -45,23 +42,32 @@ public function generateAlternative($filePath, array $format)
      *
      * @param array   $format
      * @param Imagick $image
-     * 
+     *
      * @return Imagick
      */
     protected function resizeImage(array $format, Imagick $image)
     {
-        $maxWidth = array_key_exists('max_width', $format)? $format['max_width']: -1;
-        $maxHeight = array_key_exists('max_height', $format)? $format['max_height']: -1;
+        $maxWidth = array_key_exists('max_width', $format) ? $format['max_width']: -1;
+        $maxHeight = array_key_exists('max_height', $format) ? $format['max_height']: -1;
 
         if (-2 != $maxWidth + $maxHeight) {
             $image->setimagebackgroundcolor('#000000');
             $refRatio = $maxWidth / $maxHeight;
-            $imageRatio = $image->getImageWidth() / $image->getImageHeight();
-
-            if ($refRatio > $imageRatio || $maxWidth == -1) {
-                $image = $this->resizeOnHeight($image, $maxHeight);
+            $imageRatioByHeight = $image->getImageWidth() / $image->getImageHeight();
+            $imageRatioByWidth = $image->getImageHeight() / $image->getImageWidth();
+
+            if ($refRatio > $imageRatioByHeight || $maxWidth == -1) {
+                if ($imageRatioByHeight >= 1 / $maxHeight) {
+                    $image = $this->resizeOnHeight($image, $maxHeight);
+                } else {
+                    $image = $this->liquidRescaleImageOnHeight($image, $maxHeight);
```

Comment:
why use an other method which no resize but scale picture ? 
In this case where the imageRatioByHeight < 1 why not just fix  the imageRatioByHeight to 1